src/rt/util/container/array.d

```diff
@@ -39,11 +39,11 @@ nothrow:
         size_t reqsize = mulu(T.sizeof, nlength, overflow);
         if (!overflow)
         {
-            if (nlength < length)
-                foreach (ref val; _ptr[nlength .. length]) common.destroy(val);
+            if (nlength < _length)
+                foreach (ref val; _ptr[nlength .. _length]) common.destroy(val);
             _ptr = cast(T*)common.xrealloc(_ptr, reqsize);
-            if (nlength > length)
-                foreach (ref val; _ptr[length .. nlength]) common.initialize(val);
+            if (nlength > _length)
+                foreach (ref val; _ptr[_length .. nlength]) common.initialize(val);
             _length = nlength;
```

Comment:
Wouldn't the invariant fail here if `_length` is set to 0, but `_ptr` is not set to null?